src/osd/scheduler/mClockScheduler.cc

```diff
@@ -394,12 +389,17 @@ void mClockScheduler::set_profile_config()
   update_configuration();
 }
 
-int mClockScheduler::calc_scaled_cost(int item_cost)
+uint32_t mClockScheduler::calc_scaled_cost(int item_cost)
 {
-  // Calculate total scaled cost in secs
-  int scaled_cost =
-    std::round(osd_mclock_cost_per_io + (osd_mclock_cost_per_byte * item_cost));
-  return std::max(scaled_cost, 1);
+  auto cost = static_cast<uint32_t>(
+    std::max<int>(
+      1, // ensure cost is non-zero and positive
+      item_cost));
+  auto cost_per_io = static_cast<uint32_t>(
+    std::round(osd_bandwidth_cost_per_io));
```

Comment:
IIUC we don't expect `round` will change very much here, at least for HDDs, as _bandwitch_ is expected to be in order in megabytes while _max IOPS_  – in hundreds. For SSDs the latter is ~20 k.

Reply:
Yes, the overall value won't change much. I can remove the `round` and just do a `uint32_t` conversion here.